compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/linalg_function_outlining.mlir

```diff
@@ -0,0 +1,177 @@
+// CHECK-SAME:          outs(%[[OUTPUT]] :
+// CHECK:           arith.truncf
+// CHECK:           return
+// CHECK:        }
+// CHECK-LABEL:  func.func @elemwise_example
+// CHECK-SAME:   (%[[A:.*]]: memref<4xf32>,
+// CHECK-SAME:    %[[C:.*]]: memref<4xbf16>,
+// CHECK-SAME:    %[[B:.*]]: memref<4xf32>) {
+// CHECK:            amdaie.core
+// CHECK:               func.call @generic_elementwise_0_outlined(%[[A]], %[[C]])
+// CHECK-NOT:           linalg.generic
+// CHECK:               amdaie.end
+// CHECK:            }
+// CHECK:            amdaie.core
+// CHECK:               func.call @generic_elementwise_0_outlined(%[[B]], %[[C]])
+// CHECK-NOT:           linalg.generic
+// CHECK:               amdaie.end
+// CHECK:            }
+// CHECK:            amdaie.core
+// CHECK:               func.call @generic_elementwise_1_outlined(%[[A]], %[[C]])
+// CHECK-NOT:           linalg.generic
+// CHECK:               amdaie.end
+// CHECK:            }
+// CHECK:            amdaie.core
+// CHECK:               func.call @generic_elementwise_0_outlined(%[[A]], %[[C]])
+// CHECK-NOT:           linalg.generic
+// CHECK:               amdaie.end
+// CHECK:            }
+// CHECK:            return
+// CHECK:        }
+func.func @elemwise_example(%A: memref<4xf32>, %C: memref<4xbf16>, %B: memref<4xf32>) {
+  %c2 = arith.constant 2 : index
+  %c1 = arith.constant 1 : index
+  %tile = amdaie.tile(%c1, %c2)
+  %0 = amdaie.core(%tile, in : [], out : []) {
+    linalg.generic {
+      indexing_maps = [affine_map<(d0) -> (d0)>, affine_map<(d0) -> (d0)>],
+      iterator_types = ["parallel"]
+    } ins(%A : memref<4xf32>)
+      outs(%C : memref<4xbf16>) {
+    ^bb0(%in: f32, %out: bf16):
+      %1 = arith.truncf %in : f32 to bf16
+      linalg.yield %1 : bf16
+    }
+    amdaie.end
+  }
+  %2 = amdaie.core(%tile, in : [], out : []) {
+    linalg.generic {
+      indexing_maps = [affine_map<(d0) -> (d0)>, affine_map<(d0) -> (d0)>],
+      iterator_types = ["parallel"]
+    } ins(%B : memref<4xf32>)
+      outs(%C : memref<4xbf16>) {
+    ^bb0(%in: f32, %out: bf16):
+      %3 = arith.truncf %in : f32 to bf16
+      linalg.yield %3 : bf16
+    }
+    amdaie.end
+  }
+  %4 = amdaie.core(%tile, in : [], out : []) {
+    linalg.generic {
+      indexing_maps = [affine_map<(d0) -> (d0)>, affine_map<(d0) -> (d0)>],
+      iterator_types = ["parallel"]
+    } ins(%A : memref<4xf32>)
+      outs(%C : memref<4xbf16>) {
+    ^bb0(%in: f32, %out: bf16):
+      %5 = arith.truncf %in : f32 to bf16
+      %6 = arith.addf %5, %out : bf16
+      linalg.yield %6 : bf16
+    }
+    amdaie.end
+  }
+  %7 = amdaie.core(%tile, in : [], out : []) {
+    linalg.generic {
+      indexing_maps = [affine_map<(d0) -> (d0)>, affine_map<(d0) -> (d0)>],
+      iterator_types = ["parallel"]
+    } ins(%A : memref<4xf32>)
+      outs(%C : memref<4xbf16>) {
+    ^bb0(%in: f32, %out: bf16):
+      %8 = arith.truncf %in : f32 to bf16
+      linalg.yield %8 : bf16
+    }
+    amdaie.end
+  }
```

Comment:
I think the above three elementwise cores are good enough for test purpose. Why don't you delete this or change this to a matmul.